src/application/view_photo.c

```diff
@@ -0,0 +1,465 @@
+    DestroyMsgData(viewPhoto->msgData);
+    FontID_Release(4);
+}
+
+static void ViewPhotoSysTask_InitWindows(ViewPhotoSysTaskData *viewPhoto) {
+    for (int i = 0; i < 2; ++i) {
+        AddWindow(viewPhoto->bgConfig, &viewPhoto->windows[i], &ov19_0225A04E[i]);
+        FillWindowPixelBuffer(&viewPhoto->windows[i], 0);
+    }
+}
+
+static void ViewPhotoSysTask_ReleaseWindows(ViewPhotoSysTaskData *viewPhoto) {
+    for (int i = 0; i < 2; ++i) {
+        ClearWindowTilemapAndCopyToVram(&viewPhoto->windows[i]);
+        RemoveWindow(&viewPhoto->windows[i]);
+    }
+}
+
+static void ViewPhotoSysTask_CreateSprites(ViewPhotoSysTaskData *viewPhoto) {
+    UnkFieldSpriteRenderer_ov01_021E7FDC_Init(&viewPhoto->spriteRender, ov19_0225A040, 3, viewPhoto->heapId);
+    for (int i = 0; i < 3; ++i) {
+        viewPhoto->sprites[i] = ov01_021E81F0(&viewPhoto->spriteRender, &ov19_0225A0C4[i]);
+        Set2dSpriteVisibleFlag(viewPhoto->sprites[i], TRUE);
+        Set2dSpriteAnimActiveFlag(viewPhoto->sprites[i], TRUE);
+    }
+    if (viewPhoto->scrollData.curPhoto == 0) {
+        Set2dSpriteAnimSeqNo(viewPhoto->sprites[1], 2);
+    } else {
+        Set2dSpriteAnimSeqNo(viewPhoto->sprites[1], 0);
+    }
+    if (viewPhoto->scrollData.curPhoto >= viewPhoto->scrollData.numPhotos - 1) {
+        Set2dSpriteAnimSeqNo(viewPhoto->sprites[2], 5);
+    } else {
+        Set2dSpriteAnimSeqNo(viewPhoto->sprites[2], 3);
+    }
+    GfGfx_EngineBTogglePlanes(GX_PLANEMASK_OBJ, GF_PLANE_TOGGLE_ON);
+}
+
+static void ViewPhotoSysTask_DeleteSprites(ViewPhotoSysTaskData *viewPhoto) {
+    for (int i = 0; i < 3; ++i) {
+        Sprite_Delete(viewPhoto->sprites[i]);
+    }
+    UnkFieldSpriteRenderer_ov01_021E7FDC_Release(&viewPhoto->spriteRender);
+    GfGfx_EngineBTogglePlanes(GX_PLANEMASK_OBJ, GF_PLANE_TOGGLE_OFF);
+}
+
+static void ViewPhotoSysTask_AnimateButtonSelect(ViewPhotoSysTaskData *viewPhoto, int spriteNo) {
+    viewPhoto->animSpriteNo = spriteNo;
+    Set2dSpriteAnimSeqNo(viewPhoto->sprites[spriteNo], _0225A03C[spriteNo]);
+    Sprite_ResetAnimCtrlState(viewPhoto->sprites[spriteNo]);
+}
+
+static BOOL ViewPhotoSysTask_IsButtonAnimPlaying(ViewPhotoSysTaskData *viewPhoto) {
+    return !Sprite_IsCellAnimationRunning(viewPhoto->sprites[viewPhoto->animSpriteNo]);
+}
+
+static void formatPhotoFlavorText(PHOTO *photo, MessageFormat *msgFormat, String *strBuf, HeapID heapId, SaveData *saveData) {
+    BufferPlayersName(msgFormat, 0, Save_PlayerData_GetProfileAddr(saveData));
+    sub_02068F98(photo->mapId, heapId, strBuf);
+    BufferString(msgFormat, 1, strBuf, 2, 0, 2);
+    CopyU16ArrayToString(strBuf, photo->leadMonNick);
+    BufferString(msgFormat, 2, strBuf, 2, 0, 2);
+    u8 year = photo->date >> 24;
+    BufferIntegerAsString(msgFormat, 3, year + 2000, 4, PRINTING_MODE_LEADING_ZEROS, TRUE);
+    u8 month = photo->date >> 16;
+    BufferIntegerAsString(msgFormat, 4, month, 2, PRINTING_MODE_LEADING_ZEROS, TRUE);
+    u8 day = photo->date >> 8;
```

Comment:
can't date be more easily represented as a struct?
```
typedef struct Date {
    u8 filler;
    u8 day;
    u8 month;
    u8 year;
}
```

or does it not match like that?

Reply:
the access suggests a u32 rather than a struct of four u8s